saleor/graphql/product/bulk_mutations/products.py

```diff
@@ -71,6 +90,188 @@ class Meta:
         error_type_field = "product_errors"
 
 
+class ProductVariantBulkCreateInput(ProductVariantInput):
+    attributes = graphene.List(
+        AttributeValueInput,
+        required=True,
+        description="List of attributes specific to this variant.",
+    )
+
+
+class ProductVariantBulkCreate(BaseMutation):
+    count = graphene.Int(
+        required=True,
+        default_value=0,
+        description="Returns how many objects were affected.",
+    )
+    product_variants = graphene.List(
+        ProductVariant,
+        required=True,
+        default_value=[],
+        description=(
+            "Represents a version of a products such as different size or color.",
+        ),
+    )
+
+    class Arguments:
+        variants = graphene.List(
+            ProductVariantBulkCreateInput,
+            required=True,
+            description="Fields required to create product variants.",
+        )
+        product_id = graphene.ID(
+            description="ID of the product to create the variants for",
+            name="product",
+            required=True,
+        )
+
+    class Meta:
+        description = "Creates product variants."
+        permissions = ("product.manage_products",)
+        error_type_class = BulkProductError
+        error_type_field = "bulk_product_errors"
+
+    @classmethod
+    def clean_attributes(
+        cls, attributes: dict, product_type: models.ProductType
+    ) -> T_INPUT_MAP:
+        attributes_qs = product_type.variant_attributes
+        attributes = AttributeAssignmentMixin.clean_input(
+            attributes, attributes_qs, is_variant=True
+        )
+        return attributes
+
+    @classmethod
+    def clean_input(cls, info, instance: models.ProductVariant, data: dict):
+        cleaned_input = ModelMutation.clean_input(
+            info, instance, data, ProductVariantBulkCreateInput
+        )
+
+        cost_price_amount = cleaned_input.pop("cost_price", None)
+        if cost_price_amount is not None:
+            cleaned_input["cost_price_amount"] = cost_price_amount
+
+        price_override_amount = cleaned_input.pop("price_override", None)
+        if price_override_amount is not None:
+            cleaned_input["price_override_amount"] = price_override_amount
+
+        # Attributes are provided as list of `AttributeValueInput` objects.
+        # We need to transform them into the format they're stored in the
+        # `Product` model, which is HStore field that maps attribute's PK to
+        # the value's PK.
+        attributes = cleaned_input.get("attributes")
+        if attributes:
+            try:
+                cleaned_input["attributes"] = cls.clean_attributes(
+                    attributes, data["product_type"]
+                )
+            except ValidationError as exc:
+                raise ValidationError({"attributes": exc})
+
+        return cleaned_input
```

Comment:
I would extract the logic or something as it's almost the same as `ProductVariantCreate`, which would prevent bugs.